lib/coroutine.php

```diff
@@ -100,7 +100,7 @@ function (Throwable $reason) use ($generator, $advanceGenerator) {
             if ($returnValue instanceof Promise) {
                 $returnValue->then(function ($value) use ($promise) {
                     $promise->fulfill($value);
-                }, function (Throwable $reason) {
+                }, function (Throwable $reason) use ($promise) {
```

Comment:
Ohh wow.. do we need testcoverage, since this seems to be a major error?

Reply:
It definitely suggests this :(